### ISSUE TITLE
To include or omit flagged data?

### ISSUE THREAD

**Issue by alinaCO2spera** [BLE-LTER/BLEdatatools#9]
Should product include cells with flagged data? Can this be an argument in download_data()?
How would this work for collated data?

**Comment by twhiteaker** [comment]
I typically see flagged data included, along with the flags.  I can see how that could be tricky with collated data.  What if two datasets used the same flag but with different meanings?  What if a dataset included multiple fields with different types of flags.  But if the scope is to just work with BLE data, maybe that won't be an issue.

**Comment by atn38** [comment]
Exactly @twhiteaker, thanks for visiting us. BLE already has both of your scenarios, so, food for thought. 

If we output an Excel option, the flag + flag definition can be coded into a comment on the particular cell. This will be a challenge to implement!
Not sure what other cell-level annotation options are there.

**Comment by twhiteaker** [comment]
I recommend avoiding non-vanilla features such as comments.  I think it'd be better to have the flag definition in the table or on a separate sheet.

**Comment by atn38** [comment]
good point about filtering by flag values @twhiteaker. That def tips the scales to having them in cells then.

**Comment by atn38** [comment]
working on collating data now: one issue I'm getting is that our flag-type column names are not consistent. Some data tables have "flag_N" or more descriptive names, and some just say "flag". This makes it difficult to tell what flags apply to what data in a collated setting.

**Comment by alinaCO2spera** [comment]
@atn38 and I talked about this issue today and decided we are moving forward with including all flags columns in the collated data product so users can filter by and/or remove flagged data as they wish. For the most part flag column names are not a problem. The exception is sediment pigment flags which do not have descriptive names due to the original data formatting. 

@atn38 : do you think fixing sediment pigment flag column(s) happen inside collate_data()?

**Comment by atn38** [comment]
@alinaCO2spera yea, can be direct lines of code inside collate_data, or in a helper function that then gets called in collate_data. I'm actually sorta doing something about the sediment pigment flags -- I'm gonna append the pigment name after flag and making it wider, e.g. there are a bunch of columns named "flag_Alloxanthin" "flag_Fucoxanthin" etc in the collated product now.
